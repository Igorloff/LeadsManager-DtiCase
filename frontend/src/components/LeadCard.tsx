import { toast } from "react-toastify";
import api from "../services/api";
import type { Lead } from "../types/Lead";
import styles from "./LeadCard.module.css";
import Avatar from "@mui/material/Avatar";
import Button from "@mui/material/Button";
import Divider from "@mui/material/Divider";
import { format } from "date-fns";
import {
  Email,
  Phone,
  CalendarToday,
  LocationOn,
  Numbers,
} from "@mui/icons-material";

type LeadCardProps = {
  lead: Lead;
  updateLeads: () => void;
};

export default function LeadCard({ lead, updateLeads }: LeadCardProps) {
  function formatDate(dateString: string): string {
    if (!dateString) return "";
    return format(new Date(dateString), "dd/MM/yyyy HH:mm");
  }

  const handleAccept = async () => {
    try {
      await api.post(`/Leads/${lead.id}/accept`);
      updateLeads();
      toast.success("Lead aceito com sucesso!");
      setTimeout(() => {
        toast.info(
          `Email enviado para ${lead.firstName.toLowerCase()}@gmail.com`
        );
      }, 800);
    } catch (error) {
      console.error("Erro ao aceitar lead:", error);
      toast.error("Erro ao aceitar lead!");
    }
  };

  const handleDecline = async () => {
    try {
      await api.post(`/Leads/${lead.id}/decline`);
      updateLeads();
      toast.info("Lead recusado.");
    } catch (error) {
      console.error("Erro ao recusar lead:", error);
      toast.error(" Erro ao recusar lead!");
    }
  };

  if (lead.status === 0) {
    return (
      <div className={styles.card}>
        <div className={styles.header_containers}>
          <Avatar>{lead.firstName.charAt(0)}</Avatar>
          <div className={styles.header}>
            <span className={styles.name}>{lead.firstName}</span>
            <span className={styles.category}>{lead.category}</span>
          </div>
        </div>

        <Divider sx={{ my: 1 }} />

        <div className={styles.details}>
          <p>
            <CalendarToday fontSize="small" style={{ marginRight: "6px" }} />
            {formatDate(lead.dateCreated)}
          </p>
          <p>
            <LocationOn fontSize="small" style={{ marginRight: "6px" }} />{" "}
            {lead.city}
          </p>
          <p>
            <Numbers fontSize="small" style={{ marginRight: "6px" }} />
            {lead.id}
          </p>
        </div>

        <Divider sx={{ my: -1 }} />

        <p>{lead.description}</p>

        <Divider sx={{ my: -1 }} />

        <div className={styles.last_line}>
          <div className={styles.price}> R$ {lead.price.toFixed(2)}</div>

          <div className={styles.actions}>
            <Button
              color="info"
              variant="contained"
              className={`${styles.button} ${styles.accept}`}
              onClick={handleAccept}
            >
              Accept
            </Button>
            <Button
              color="warning"
              variant="contained"
              className={`${styles.button} ${styles.decline}`}
              onClick={handleDecline}
            >
              Decline
            </Button>
          </div>
        </div>
      </div>
    );
  }
  return (
    <div className={styles.card}>
      <div className={styles.header_containers}>
        <Avatar>{lead.firstName.charAt(0)}</Avatar>
        <div className={styles.header}>
          <span className={styles.name}>{lead.fullName}</span>
          <span className={styles.category}>{lead.category}</span>
        </div>
      </div>

      <Divider sx={{ my: 1 }} />

      <div className={styles.details}>
        <p>
          <CalendarToday fontSize="small" style={{ marginRight: "6px" }} />
          {formatDate(lead.dateCreated)}
        </p>
        <p>
          <LocationOn fontSize="small" style={{ marginRight: "6px" }} />{" "}
          {lead.city}
        </p>
        <p>
          <Numbers fontSize="small" style={{ marginRight: "6px" }} />
          {lead.id}
        </p>
      </div>

      <Divider sx={{ my: -1 }} />

      <div className={styles.additional_details}>
        <p>
          <Phone fontSize="small" style={{ marginRight: "6px" }} />{" "}
          {lead.contactPhone}
        </p>
        <p>
          <Email fontSize="small" style={{ marginRight: "6px" }} />
          {lead.email}
        </p>
      </div>

      <Divider sx={{ my: -1 }} />

      <p>{lead.description}</p>

      <Divider sx={{ my: -1 }} />

      <div className={styles.last_line}>
        <div className={styles.price}> R$ {lead.price.toFixed(2)}</div>
      </div>
    </div>
  );
}
