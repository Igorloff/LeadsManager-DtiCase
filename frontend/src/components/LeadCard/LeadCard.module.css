.card {
  background: #fff;
  border-radius: 8px;
  box-shadow: 0 4px 12px rgba(0, 0, 0, 0.08);
  display: flex;
  flex-direction: column;
  gap: 12px;
  padding: 20px;
  transition: transform 0.2s ease, box-shadow 0.2s ease;
  color: #444;
}

.card:hover {
  transform: translateY(-4px);
  box-shadow: 0 8px 20px rgba(0, 0, 0, 0.2);
}

.header_containers {
  display: grid;
  grid-template-columns: 1fr 10fr;
}

.header {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.name {
  font-size: 1.2rem;
  font-weight: 600;
  color: #333;
}

.category {
  justify-content: center;
  font-size: 0.9rem;
  font-weight: 500;
  color: #ffffff;
  background: #007bff;
  padding: 4px 12px;
  border-radius: 12px;
}

.details {
  display: flex;
  gap: 80px;
  font-size: 0.95rem;
  line-height: 1.5;
}

.additional_details {
  display: flex;
  font-weight: 500;
  color: rgb(0, 59, 185);
  gap: 80px;
  font-size: 0.95rem;
  line-height: 1.5;
}

.last_line {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.price {
  font-weight: 600;
  color: #2a9d8f;
  font-size: 1.1rem;
}

.actions {
  display: flex;
  justify-content: flex-end;
  gap: 40px;
}

.button {
  width: 120px;
  padding: 8px 14px;
  border-radius: 8px;
  font-size: 0.9rem;
  font-weight: 500;
  border: none;
  cursor: pointer;
  transition: background 0.2s ease;
}

.accept {
  background: #2a9d8f;
  color: white;
}

.accept:hover {
  background: #23877a;
}

.decline {
  background: #e63946;
  color: white;
}

.decline:hover {
  background: #c92c3a;
}
