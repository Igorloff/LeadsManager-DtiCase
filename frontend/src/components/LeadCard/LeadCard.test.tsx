import { expect, test, it, vi, describe } from "vitest";
import { render, fireEvent, waitFor } from "@testing-library/react";
import LeadCard from "./LeadCard";
import type { Lead } from "../../types/Lead";

vi.mock("axios", () => {
  return {
    default: {
      post: vi.fn(),
      get: vi.fn(),
      delete: vi.fn(),
      put: vi.fn(),
      create: vi.fn().mockReturnThis(),
      interceptors: {
        request: {
          use: vi.fn(),
          eject: vi.fn(),
        },
        response: {
          use: vi.fn(),
          eject: vi.fn(),
        },
      },
    },
  };
});

const leadMock: Lead = {
  id: 1,
  firstName: "John",
  lastName: "Doe",
  fullName: "John Doe",
  contactPhone: "123456789",
  email: "eareis@gmail.com",
  city: "New York",
  category: "Real Estate",
  description: "Looking for a 2-bedroom apartment in downtown.",
  price: 2500,
  dateCreated: "2023-10-01T10:00:00Z",
  status: 0,
};

const mockFn = vi.fn();

describe("LeadCard", () => {
  it("Aceita um LeadCard", async () => {
    const container = render(<LeadCard lead={leadMock} updateLeads={mockFn} />);
    const acceptButton = container.getByText("Accept");

    expect(acceptButton).toBeDefined();

    await waitFor(() => {
      fireEvent.click(acceptButton);
      expect(mockFn).toHaveBeenCalled();
    });

    container.unmount();
  });

  it("Rejeita um LeadCard", async () => {
    const container = render(<LeadCard lead={leadMock} updateLeads={mockFn} />);
    const declineButton = container.getByText("Decline");

    expect(declineButton).toBeDefined();

    await waitFor(() => {
      fireEvent.click(declineButton);
      expect(mockFn).toHaveBeenCalled();
    });

    container.unmount();
  });

  it("Rendezira LeadCard com status 0", () => {
    const container = render(<LeadCard lead={leadMock} updateLeads={mockFn} />);
    expect(container).toMatchSnapshot();

    container.unmount();
  });

  it("Rendezira LeadCard com status 1", () => {
    const container = render(
      <LeadCard lead={{ ...leadMock, status: 1 }} updateLeads={mockFn} />
    );
    expect(container).toMatchSnapshot();

    container.unmount();
  });
});
