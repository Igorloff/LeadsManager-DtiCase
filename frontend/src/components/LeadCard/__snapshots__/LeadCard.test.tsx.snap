// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LeadCard > Rendezira LeadCard com status 0 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div />
    <div>
      <div
        class="_card_4caed2"
      >
        <div
          class="_header_containers_4caed2"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-18k2bs-MuiAvatar-root"
          >
            J
          </div>
          <div
            class="_header_4caed2"
          >
            <span
              class="_name_4caed2"
            >
              John
            </span>
            <span
              class="_category_4caed2"
            >
              Real Estate
            </span>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-hv6zls-MuiDivider-root"
        />
        <div
          class="_details_4caed2"
        >
          <p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
              data-testid="CalendarTodayIcon"
              focusable="false"
              style="margin-right: 6px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m0 18H4V8h16z"
              />
            </svg>
            01/10/2023 07:00
          </p>
          <p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
              data-testid="LocationOnIcon"
              focusable="false"
              style="margin-right: 6px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7m0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5"
              />
            </svg>
             
            New York
          </p>
          <p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
              data-testid="NumbersIcon"
              focusable="false"
              style="margin-right: 6px;"
              viewBox="0 0 24 24"
            >
              <path
                d="m20.5 10 .5-2h-4l1-4h-2l-1 4h-4l1-4h-2L9 8H5l-.5 2h4l-1 4h-4L3 16h4l-1 4h2l1-4h4l-1 4h2l1-4h4l.5-2h-4l1-4zm-7 4h-4l1-4h4z"
              />
            </svg>
            1
          </p>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-180woc-MuiDivider-root"
        />
        <p>
          Looking for a 2-bedroom apartment in downtown.
        </p>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-180woc-MuiDivider-root"
        />
        <div
          class="_last_line_4caed2"
        >
          <div
            class="_price_4caed2"
          >
             R$ 
            2500.00
          </div>
          <div
            class="_actions_4caed2"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInfo MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInfo _button_4caed2 _accept_4caed2 css-mnflth-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Accept
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedWarning MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorWarning MuiButton-root MuiButton-contained MuiButton-containedWarning MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorWarning _button_4caed2 _decline_4caed2 css-14169-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Decline
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_card_4caed2"
    >
      <div
        class="_header_containers_4caed2"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-18k2bs-MuiAvatar-root"
        >
          J
        </div>
        <div
          class="_header_4caed2"
        >
          <span
            class="_name_4caed2"
          >
            John
          </span>
          <span
            class="_category_4caed2"
          >
            Real Estate
          </span>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-hv6zls-MuiDivider-root"
      />
      <div
        class="_details_4caed2"
      >
        <p>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
            data-testid="CalendarTodayIcon"
            focusable="false"
            style="margin-right: 6px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m0 18H4V8h16z"
            />
          </svg>
          01/10/2023 07:00
        </p>
        <p>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
            data-testid="LocationOnIcon"
            focusable="false"
            style="margin-right: 6px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7m0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5"
            />
          </svg>
           
          New York
        </p>
        <p>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
            data-testid="NumbersIcon"
            focusable="false"
            style="margin-right: 6px;"
            viewBox="0 0 24 24"
          >
            <path
              d="m20.5 10 .5-2h-4l1-4h-2l-1 4h-4l1-4h-2L9 8H5l-.5 2h4l-1 4h-4L3 16h4l-1 4h2l1-4h4l-1 4h2l1-4h4l.5-2h-4l1-4zm-7 4h-4l1-4h4z"
            />
          </svg>
          1
        </p>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-180woc-MuiDivider-root"
      />
      <p>
        Looking for a 2-bedroom apartment in downtown.
      </p>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-180woc-MuiDivider-root"
      />
      <div
        class="_last_line_4caed2"
      >
        <div
          class="_price_4caed2"
        >
           R$ 
          2500.00
        </div>
        <div
          class="_actions_4caed2"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInfo MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInfo _button_4caed2 _accept_4caed2 css-mnflth-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Accept
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedWarning MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorWarning MuiButton-root MuiButton-contained MuiButton-containedWarning MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorWarning _button_4caed2 _decline_4caed2 css-14169-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Decline
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`LeadCard > Rendezira LeadCard com status 1 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div />
    <div />
    <div>
      <div
        class="_card_4caed2"
      >
        <div
          class="_header_containers_4caed2"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-18k2bs-MuiAvatar-root"
          >
            J
          </div>
          <div
            class="_header_4caed2"
          >
            <span
              class="_name_4caed2"
            >
              John Doe
            </span>
            <span
              class="_category_4caed2"
            >
              Real Estate
            </span>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-hv6zls-MuiDivider-root"
        />
        <div
          class="_details_4caed2"
        >
          <p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
              data-testid="CalendarTodayIcon"
              focusable="false"
              style="margin-right: 6px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m0 18H4V8h16z"
              />
            </svg>
            01/10/2023 07:00
          </p>
          <p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
              data-testid="LocationOnIcon"
              focusable="false"
              style="margin-right: 6px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7m0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5"
              />
            </svg>
             
            New York
          </p>
          <p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
              data-testid="NumbersIcon"
              focusable="false"
              style="margin-right: 6px;"
              viewBox="0 0 24 24"
            >
              <path
                d="m20.5 10 .5-2h-4l1-4h-2l-1 4h-4l1-4h-2L9 8H5l-.5 2h4l-1 4h-4L3 16h4l-1 4h2l1-4h4l-1 4h2l1-4h4l.5-2h-4l1-4zm-7 4h-4l1-4h4z"
              />
            </svg>
            1
          </p>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-180woc-MuiDivider-root"
        />
        <div
          class="_additional_details_4caed2"
        >
          <p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
              data-testid="PhoneIcon"
              focusable="false"
              style="margin-right: 6px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02z"
              />
            </svg>
             
            123456789
          </p>
          <p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
              data-testid="EmailIcon"
              focusable="false"
              style="margin-right: 6px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2m0 4-8 5-8-5V6l8 5 8-5z"
              />
            </svg>
            eareis@gmail.com
          </p>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-180woc-MuiDivider-root"
        />
        <p>
          Looking for a 2-bedroom apartment in downtown.
        </p>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-180woc-MuiDivider-root"
        />
        <div
          class="_last_line_4caed2"
        >
          <div
            class="_price_4caed2"
          >
             R$ 
            2500.00
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_card_4caed2"
    >
      <div
        class="_header_containers_4caed2"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-18k2bs-MuiAvatar-root"
        >
          J
        </div>
        <div
          class="_header_4caed2"
        >
          <span
            class="_name_4caed2"
          >
            John Doe
          </span>
          <span
            class="_category_4caed2"
          >
            Real Estate
          </span>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-hv6zls-MuiDivider-root"
      />
      <div
        class="_details_4caed2"
      >
        <p>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
            data-testid="CalendarTodayIcon"
            focusable="false"
            style="margin-right: 6px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m0 18H4V8h16z"
            />
          </svg>
          01/10/2023 07:00
        </p>
        <p>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
            data-testid="LocationOnIcon"
            focusable="false"
            style="margin-right: 6px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7m0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5"
            />
          </svg>
           
          New York
        </p>
        <p>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
            data-testid="NumbersIcon"
            focusable="false"
            style="margin-right: 6px;"
            viewBox="0 0 24 24"
          >
            <path
              d="m20.5 10 .5-2h-4l1-4h-2l-1 4h-4l1-4h-2L9 8H5l-.5 2h4l-1 4h-4L3 16h4l-1 4h2l1-4h4l-1 4h2l1-4h4l.5-2h-4l1-4zm-7 4h-4l1-4h4z"
            />
          </svg>
          1
        </p>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-180woc-MuiDivider-root"
      />
      <div
        class="_additional_details_4caed2"
      >
        <p>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
            data-testid="PhoneIcon"
            focusable="false"
            style="margin-right: 6px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02z"
            />
          </svg>
           
          123456789
        </p>
        <p>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
            data-testid="EmailIcon"
            focusable="false"
            style="margin-right: 6px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2m0 4-8 5-8-5V6l8 5 8-5z"
            />
          </svg>
          eareis@gmail.com
        </p>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-180woc-MuiDivider-root"
      />
      <p>
        Looking for a 2-bedroom apartment in downtown.
      </p>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-180woc-MuiDivider-root"
      />
      <div
        class="_last_line_4caed2"
      >
        <div
          class="_price_4caed2"
        >
           R$ 
          2500.00
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
